From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Andreas Dinter <andreas.dinter@systec-electronic.com>
Date: Sun, 2 Mar 2025 20:28:06 +0100
Subject: [PATCH] drivers/gpio/gpio-aggregator.c: Add compatible
 "gpio-aggregator"

Signed-off-by: Andreas Dinter <andreas.dinter@systec-electronic.com>
---
 drivers/gpio/gpio-aggregator.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/gpio/gpio-aggregator.c b/drivers/gpio/gpio-aggregator.c
index cc6ee4334602..f23cb9af7f7d 100644
--- a/drivers/gpio/gpio-aggregator.c
+++ b/drivers/gpio/gpio-aggregator.c
@@ -619,6 +619,10 @@ static const struct of_device_id gpio_aggregator_dt_ids[] = {
 	 * Add GPIO-operated devices controlled from userspace below,
 	 * or use "driver_override" in sysfs.
 	 */
+	{
+		.compatible = "gpio-aggregator",
+		.data = NULL,
+	},
 	{}
 };
 MODULE_DEVICE_TABLE(of, gpio_aggregator_dt_ids);
-- 
2.52.0

