From ed09d25ed6f505e9b3fb9cb3d60e74ed5ba28d4c Mon Sep 17 00:00:00 2001
From: Andreas Dinter <andreas.dinter@systec-electronic.com>
Date: Thu, 8 May 2025 14:13:25 +0200
Subject: [PATCH] Cargo.toml: Remove workspace member ctr700drv

Signed-off-by: Andreas Dinter <andreas.dinter@systec-electronic.com>
---
 Cargo.toml | 1 -
 1 file changed, 1 deletion(-)

diff --git a/Cargo.toml b/Cargo.toml
index 0faec09..6a7d18c 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -11,7 +11,6 @@ resolver = "2"
 [workspace]
 members = [
     # ".",
-    "ctr700drv",
     "Bindings/sysworxx-io-js",
     "Bindings/Codesys",
     "Bindings/CSharp/SysworxxIoCsbindgen",
-- 
2.49.0

