From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Andreas Dinter <andreas.dinter@systec-electronic.com>
Date: Mon, 11 Aug 2025 14:37:51 +0200
Subject: [PATCH] board/ti/am62x/sysworxx.c: Remove obsolete board detection

Signed-off-by: Andreas Dinter <andreas.dinter@systec-electronic.com>
---
 board/ti/am62x/sysworxx.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/board/ti/am62x/sysworxx.c b/board/ti/am62x/sysworxx.c
index aa754e4722b..dba5a1ece05 100644
--- a/board/ti/am62x/sysworxx.c
+++ b/board/ti/am62x/sysworxx.c
@@ -127,7 +127,6 @@ void reset_phy(void) {
 #ifdef CONFIG_BOARD_LATE_INIT
 int board_late_init(void)
 {
-	ti_set_fdt_env(NULL, NULL);
 	return 0;
 }
 #endif
-- 
2.50.1

